scipy/sparse/linalg/isolve/lsmr.py

```diff
@@ -171,15 +175,22 @@ def lsmr(A, b, damp=0.0, atol=1e-6, btol=1e-6, conlim=1e8,
         print('btol = %8.2e             maxiter = %8g\n' % (btol, maxiter))
 
     u = b
-    beta = norm(u)
-
-    v = zeros(n)
-    alpha = 0
+    normb = norm(b)
+    if x0 is None:
+        x = zeros(n)
+        beta = normb.copy()
+    else:
+        x = atleast_1d(x0)
+        u -= A.matvec(x)
```

Comment:
yes, versioning mixup on my part.  Sorry.  It's fixed and waiting for the tests to pass.